### PR TITLE
chirp: modernize the expression by using buildPythonApplication

### PR DESCRIPTION
**Pull request by andir**
###### Motivation for this change

I was asked to have a look on this since it apparently broke some time ago.

###### Things done
I only built the application. Testing has to be done by someone that actually own and uses HAM radio equipment (cc @hax404).

### CODE REVIEW

**Comment by worldofpeace** [comment]
This ties in with the work in #74295. It doesn't seem there's a port https://chirp.danplanet.com/issues/6327. However it's not a very strong contender for removal, there's actual activity and users https://chirp.danplanet.com/projects/chirp/repository/revisions.

File: pkgs/applications/radio/chirp/default.nix [hunk 1]
```diff
@@ -1,31 +1,20 @@
-{ stdenv, fetchurl, libxml2Python, libxslt, makeWrapper
-, pyserial, pygtk }:
-
-stdenv.mkDerivation rec {
+{ stdenv
+, fetchurl
+, python2
```
Comment (FRidh) [inline pkgs/applications/radio/chirp/default.nix]: Does it not work with Python 3?

Comment (worldofpeace) [reply pkgs/applications/radio/chirp/default.nix]: my best guess is there's a branch with it https://chirp.danplanet.com/projects/chirp/wiki/Linux_Python3.

Comment (worldofpeace) [reply pkgs/applications/radio/chirp/default.nix]: yep https://chirp.danplanet.com/projects/chirp/repository/show?rev=py3

Comment (andir) [reply pkgs/applications/radio/chirp/default.nix]: The port isn't really complete and has a bunch of issues. Last I checked the first thing I saw was incompatible exception syntax.

**Comment by hax404** [comment]
Tested programming of Kenwood TH-D72 and Baofeng GT-3 Radio. Works.

**Pull request merged**